prow/cmd/generic-autobumper/bumper/bumper_test.go

```diff
@@ -786,6 +804,16 @@ func TestUpstreamImageVersionResolver(t *testing.T) {
 			currentTag:          "whatever-current-tag",
 			expectedTargetTag:   "whatever-current-tag",
 		},
+		{
+			description:         "tag does not exist",
+			upstreamVersionType: upstreamVersion,
+			expectError:         false,
+			prefixes:            []Prefix{tagDoesNotExistPrefix},
+			imageHost:           doesNotExistPrefix,
+			currentTag:          "doesNotExist",
+			expectedTargetTag:   "",
+			resolverError:       true,
```

Comment:
Note that while we do generate an error from the tagPicker in this case, the error is just printed to the log. https://github.com/kubernetes/test-infra/blob/f7e21a3c18f4f4bbc7ee170675ed53e4544a0632/experiment/image-bumper/bumper/bumper.go#L221
This will not cause the job to fail so this will be kind of invisible to users. Is that the preferred behavior here? I think it would be nice to know that an image couldn't be bumped, but we won't want to block the autobumper on that in all cases. Some sort of excusable notification would be nice, but I don't think we have a suitable mechanism. This might be the best option for now, just something to consider.

Reply:
Yea I noticed this as well. It helps that If "consistentImages" is set, then being unable to bump one of the images will cause the autobump job to fail because the images will not be consistent after the bump. The user would still need to go through the logs to find the specific cause though. I figured this was best because it would not interfere with bumps where we don't care if the images are consistent.